src/compute-client/src/protocol/history.rs

```diff
@@ -98,6 +98,7 @@ where
         let mut final_configuration = ComputeParameters::default();
 
         let mut initialization_complete = false;
+        let mut read_only = true;
```

Comment:
This is where I most strongly (but still not very strongly) feel that `allow_writes` would be a better name for the variable.